src/bhv_basic_move.cpp

```diff
@@ -105,21 +106,29 @@ Bhv_BasicMove::execute( PlayerAgent * agent )
     agent->debugClient().setTarget( target_point );
     agent->debugClient().addCircle( target_point, dist_thr );
 
-    if ( ! Body_GoToPoint( target_point, dist_thr, dash_power
-                           ).execute( agent ) )
-    {
-        Body_TurnToBall().execute( agent );
+    bool success = false;
+    if (ball.posValid() || wm.self().collidesWithPost()) {
+      success = true;
     }
 
-    if ( wm.existKickableOpponent()
-         && wm.ball().distFromSelf() < 18.0 )
-    {
-        agent->setNeckAction( new Neck_TurnToBall() );
+    if ( Body_GoToPoint( target_point, dist_thr, dash_power
```

Comment:
Sorry, I was unclear. How about we structure this as follows:

bool success = false;
if ( Body_GoToPoint(..) || Body_TurnToBall(...)) {
  if (ball.posValid || wm. self.collideswithPost()) {
    success = true;
  }
}